ecommerce/views/v0/__init__.py

```diff
@@ -2,97 +2,127 @@
+    DiscountSerializer,
+    OrderHistorySerializer,
+    OrderSerializer,
+    ProductSerializer,
+    UserDiscountMetaSerializer,
+    UserDiscountSerializer,
+)
+from flexiblepricing.api import determine_courseware_flexible_price_discount
+from flexiblepricing.models import FlexiblePriceTier
+from flexiblepricing.serializers import FlexiblePriceTierSerializer
 from main.constants import (
+    USER_MSG_TYPE_ENROLL_BLOCKED,
     USER_MSG_TYPE_PAYMENT_ACCEPTED,
     USER_MSG_TYPE_PAYMENT_CANCELLED,
     USER_MSG_TYPE_PAYMENT_DECLINED,
     USER_MSG_TYPE_PAYMENT_ERROR,
     USER_MSG_TYPE_PAYMENT_ERROR_UNKNOWN,
     USER_MSG_TYPE_PAYMENT_REVIEW,
-    USER_MSG_TYPE_ENROLL_BLOCKED,
 )
+from main.settings import ECOMMERCE_DEFAULT_PAYMENT_GATEWAY
+from main.utils import redirect_with_user_message
+from main.views import RefinePagination
+from users.models import User
 
-from django.views.generic import TemplateView, RedirectView, View
-from django.http import HttpResponse, Http404
-from django.shortcuts import render
-from django.utils.decorators import method_decorator
-from django.views.decorators.csrf import csrf_exempt
-from django.contrib.auth.mixins import LoginRequiredMixin
-from django.db import transaction
-from django.db.models import Q, Count
-from django.core.exceptions import ObjectDoesNotExist, PermissionDenied
-from django.urls import reverse
+log = logging.getLogger(__name__)
 
-from mitol.common.utils import now_in_utc
-from rest_framework_extensions.mixins import NestedViewSetMixin
-from mitol.payment_gateway.api import PaymentGateway
 
-from courses.models import (
-    CourseRun,
-    Course,
-    Program,
-    ProgramRun,
-)
+class ProductsPagination(RefinePagination):
+    default_limit = 2
 
-from ecommerce import api
-from ecommerce.discounts import DiscountType
-from ecommerce.serializers import (
-    OrderHistorySerializer,
-    ProductSerializer,
-    BasketSerializer,
-    BasketItemSerializer,
-    BasketDiscountSerializer,
-    BasketWithProductSerializer,
-    OrderSerializer,
-    DiscountSerializer,
-    DiscountRedemptionSerializer,
-    DiscountProductSerializer,
-    UserDiscountSerializer,
-    UserDiscountMetaSerializer,
-)
-from ecommerce.models import (
-    Product,
-    Basket,
-    BasketItem,
-    Discount,
-    DiscountRedemption,
-    DiscountProduct,
-    BasketDiscount,
-    UserDiscount,
-    Order,
-)
 
-from flexiblepricing.api import determine_courseware_flexible_price_discount
+class AllProductViewSet(ModelViewSet):
+    """This doesn't filter unenrollable products out, and adds name search for
+    courseware object readable id. It's really for the staff dashboard."""
 
-log = logging.getLogger(__name__)
+    serializer_class = ProductSerializer
+    pagination_class = ProductsPagination
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [IsAdminUser]
 
+    def get_queryset(self):
+        name_search = self.request.query_params.get("q")
```

Comment:
We need to update the query param name, We are using `search` at the frontend so this search is not working while assigning products.